### PR TITLE
HA Group object implementation

### PR DESCRIPTION
**Pull request by venkat-iblox**

### CODE REVIEW

File: examples/resources/bloxone_ipam_ha_group/resource.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+data "bloxone_ipam_dhcp_hosts" "example_host_1" {
+  filter = {
+    name = "TF_TEST_HOST_01"
```
Comment (mathewab) [inline examples/resources/bloxone_ipam_ha_group/resource.tf:3]: for the examples you can avoid the TF_TEST host names

File: examples/resources/bloxone_ipam_ha_group/resource.tf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+data "bloxone_ipam_dhcp_hosts" "example_host_2" {
+  filter = {
+    name = "TF_TEST_HOST_02"
+  }
+}
+
+resource "bloxone_ipam_ha_group" "example" {
+  name = "example_ha_group"
+}
+
```
Comment (mathewab) [inline examples/resources/bloxone_ipam_ha_group/resource.tf:16]: can you create an HA with no hosts, I thought the hosts are mandatory.

File: internal/service/ipam/api_ha_group_resource.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	return &HaGroupResource{}
+}
+
+// HaGroupResource defines the resource implementation.
+type HaGroupResource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (r *HaGroupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "ipam_ha_group"
```
Comment (mathewab) [inline internal/service/ipam/api_ha_group_resource.go:29]: The name should be `dhcp_ha_group` as the API endpoint is dhcp/ha_group

File: internal/service/ipam/model_ipamsvc_ha_group_host.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	"heartbeats": types.ListType{ElemType: types.ObjectType{AttrTypes: IpamsvcHAGroupHeartbeatsAttrTypes}},
+	"host":       types.StringType,
+	"port":       types.Int64Type,
+	"role":       types.StringType,
+	"state":      types.StringType,
+}
+
+var IpamsvcHAGroupHostResourceSchemaAttributes = map[string]schema.Attribute{
+	"address": schema.StringAttribute{
+		Computed:            true,
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_ha_group_host.go:40]: This is not a read only field and can be provided by an user. It should have both Optional and Computed as true

File: internal/service/ipam/model_ipamsvc_ha_group_host.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	"role": schema.StringAttribute{
+		Required: true,
+		Validators: []validator.String{
+			stringvalidator.OneOf("active", "passive"),
+		},
+		MarkdownDescription: "The role of this host in the HA relationship: _active_ or _passive_.",
+	},
+	"state": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "The state of DHCP on the host. This field is set when the _collect_stats_ is set to _true_ in the _GET_ _/dhcp/ha_group_ request.",
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_ha_group_host.go:67]: are we setting `collect_stats` as true anywhere in our requests, or can the user provide it?

File: internal/service/ipam/model_ipamsvc_ha_group.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+var IpamsvcHAGroupResourceSchemaAttributes = map[string]schema.Attribute{
+	"anycast_config_id": schema.StringAttribute{
+		Optional:            true,
+		MarkdownDescription: "The resource identifier.",
+	},
+	"comment": schema.StringAttribute{
+		Optional: true,
+		Validators: []validator.String{
+			stringvalidator.LengthBetween(1, 1024),
```
Comment (mathewab) [inline internal/service/ipam/model_ipamsvc_ha_group.go:59]:
```suggestion
			stringvalidator.LengthBetween(0, 1024),
```

Description says, may contain 0 chars as well :)

File: examples/resources/bloxone_ipam_ha_group/resource.tf (hunk 2 above)
Comment (venkat-iblox) [reply examples/resources/bloxone_ipam_ha_group/resource.tf:16]: You are correct, I should remove this

File: internal/service/ipam/model_ipamsvc_ha_group_host.go (hunk 4 above)
Comment (venkat-iblox) [reply internal/service/ipam/model_ipamsvc_ha_group_host.go:40]: No, the address has to be a computed field. The value is retrieved by the API from the `hosts` table.

File: internal/service/ipam/model_ipamsvc_ha_group.go (hunk 6 above)
Comment (venkat-iblox) [reply internal/service/ipam/model_ipamsvc_ha_group.go:59]: I will fix that. Wondering how the acceptance tests passed locally with this change now :)

File: internal/service/ipam/model_ipamsvc_ha_group_host.go (hunk 5 above)
Comment (venkat-iblox) [reply internal/service/ipam/model_ipamsvc_ha_group_host.go:67]: Something the user provides on the `list` and `get` calls. Let me check if I have to make changes to the generator and go-client repo to bring this field.

Comment (venkat-iblox) [reply internal/service/ipam/model_ipamsvc_ha_group_host.go:67]: This is particularly useful for clients who have a large set of HA Groups to build automation to monitor the status/state of HA using Terraform.

File: internal/service/ipam/model_ipamsvc_ha_group_host.go (hunk 4 above)
Comment (mathewab) [reply internal/service/ipam/model_ipamsvc_ha_group_host.go:40]: You have the option to change that address in the UI.

File: internal/provider/provider.go [hunk 7]
```diff
@@ -87,7 +87,8 @@ func (p *BloxOneProvider) Resources(ctx context.Context) []func() resource.Resou
 		ipam.NewAddressResource,
 		ipam.NewRangeResource,
 		ipam.NewFixedAddressResource,
-		ipam.NewServerResource,
+    ipam.NewServerResource,
+		ipam.NewHaGroupResource,
```
Comment (mathewab) [inline internal/provider/provider.go:91]: Looks like a formatting issue. Have you run gofmt?

Comment (venkat-iblox) [reply internal/provider/provider.go:91]: Ahh, no actually. I fixed the conflicts using the Github UI.

**Review by mathewab — APPROVED** [review]

**Pull request merged**